src/Raven.Server/Documents/ETL/Providers/Queue/AzureQueueStorage/AzureQueueStorageDocumentTransformer.cs

```diff
@@ -0,0 +1,106 @@
+        LoadToFunction(queueName, document, null);
+    }
+
+    private void LoadToFunction(string queueName, ScriptRunnerResult document, CloudEventAttributes attributes)
+    {
+        if (queueName == null)
+            ThrowLoadParameterIsMandatory(nameof(queueName));
+
+        var result = document.TranslateToObject(Context);
+
+        var topic = GetOrAdd(queueName);
+
+        topic.Items.Add(new AzureQueueStorageItem(Current) { TransformationResult = result, Attributes = attributes });
+    }
+
+    public override void Initialize(bool debugMode)
+    {
+        base.Initialize(debugMode);
+
+        DocumentScript.ScriptEngine.SetValue(Transformation.LoadTo,
+            new ClrFunction(DocumentScript.ScriptEngine, Transformation.LoadTo,
+                LoadToFunctionTranslatorWithAttributes));
+
+        foreach (var queueName in LoadToDestinations)
+        {
+            var name = Transformation.LoadTo + queueName;
+
+            DocumentScript.ScriptEngine.SetValue(name, new ClrFunction(DocumentScript.ScriptEngine, name,
+                (self, args) => LoadToFunctionTranslatorWithAttributes(queueName, args)));
+        }
+    }
+
+    private JsValue LoadToFunctionTranslatorWithAttributes(JsValue self, JsValue[] args)
+    {
+        var methodSignature = "loadTo(name, obj, attributes)";
+
+        if (args.Length != 2 && args.Length != 3)
+            ThrowInvalidScriptMethodCall($"{methodSignature} must be called with 2 or 3 parameters");
+
+        if (args[0].IsString() == false)
+            ThrowInvalidScriptMethodCall($"{methodSignature} first argument must be a string");
+
+        if (args[1].IsObject() == false)
+            ThrowInvalidScriptMethodCall($"{methodSignature} second argument must be an object");
+
+        if (args.Length == 3 && args[2].IsObject() == false)
+            ThrowInvalidScriptMethodCall($"{methodSignature} third argument must be an object");
+
+        return LoadToFunctionTranslatorWithAttributesInternal(args[0].AsString(), args[1].AsObject(),
+            args.Length == 3 ? args[2].AsObject() : null);
+    }
+
+    private JsValue LoadToFunctionTranslatorWithAttributes(string name, JsValue[] args)
+    {
+        var methodSignature = $"loadTo{name}(obj, attributes)";
+
+        if (args.Length != 1 && args.Length != 2)
+            ThrowInvalidScriptMethodCall($"{methodSignature} must be called with with 1 or 2 parameters");
+
+        if (args[0].IsObject() == false)
+            ThrowInvalidScriptMethodCall($"{methodSignature} argument 'obj' must be an object");
+
+        if (args.Length == 2 && args[1].IsObject() == false)
+            ThrowInvalidScriptMethodCall($"{methodSignature} argument 'attributes' must be an object");
+
+        return LoadToFunctionTranslatorWithAttributesInternal(name, args[0].AsObject(),
+            args.Length == 2 ? args[1].AsObject() : null);
+    }
+
+    private JsValue LoadToFunctionTranslatorWithAttributesInternal(string name, ObjectInstance obj,
```

Comment:
There is a lot of code duplication here. Do we need it? IIRC only RabbitMQ ETL has specific syntax. Kafka and Azure Queue will have the same

Reply:
It will be handled separately

Reply:
Can we do something about that?